File: content/problems/0102-binary-tree-level-order-traversal/solutions/bfs/solution.cpp
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        if (!root) return {};
        vector<vector<int>> result;
        queue<TreeNode*> nodes;
        nodes.push(root);
        while (!nodes.empty()) {
            int size = nodes.size();
            vector<int> level;
            while (size--) {
                TreeNode* node = nodes.front(); nodes.pop();
                level.push_back(node->val);
                if (node->left) nodes.push(node->left);
                if (node->right) nodes.push(node->right);
            }
            result.push_back(move(level));
        }
        return result;
    }
};

File: content/problems/0102-binary-tree-level-order-traversal/solutions/bfs/Solution.kt
class Solution {
    fun levelOrder(root: TreeNode?): List<List<Int>> {
        if (root == null) return emptyList()
        val result = mutableListOf<List<Int>>()
        val queue = java.util.ArrayDeque<TreeNode>()
        queue.offer(root)
        while (queue.isNotEmpty()) {
            val level = mutableListOf<Int>()
            repeat(queue.size) {
                val node = queue.poll()
                level.add(node.`val`)
                node.left?.let(queue::offer)
                node.right?.let(queue::offer)
            }
            result.add(level)
        }
        return result
    }
}
